kedro/config/omegaconf_config.py

```diff
@@ -233,9 +233,8 @@ def load_and_merge_dir_config(self, conf_path: str, patterns: Iterable[str]):
 
         if aggregate_config:
             if len(aggregate_config) > 1:
-                merged_config = OmegaConf.merge(*aggregate_config)
-                return OmegaConf.to_container(merged_config)
-            return OmegaConf.to_container(list(aggregate_config)[0])
+                return dict(OmegaConf.merge(*aggregate_config))
+            return list(aggregate_config)[0]
         return {}
```

Comment:
Could we flatten this up a bit? Nested ifs are harder to read, so maybe we can do something like:
```
if not aggregate_config:
    return {}
if len(aggregate_config)==1:
    return list(aggregate_config)[0]
return dict(OmegaConf.merge(*aggregate_config))
```

Reply:
Done!